controllers/controller.go

```diff
@@ -156,3 +161,178 @@ func Descrypt_mam_response(mam_message string, private_key string) Endpoint_obd2
 	log.Println("Descrypt mam response successfully")
 	return endpoint_data
 }
+
+func Register_device(c *gin.Context) {
+	var regDevice device.Device
+	if err := c.BindJSON(&regDevice); err != nil {
+		c.JSON(http.StatusOK, gin.H{
```

Comment:
Should you add logger in error handling?

Reply:
No, this is not the error. The user should post the correct json format.